### PR TITLE
Add histogram support.

### PR DESCRIPTION
**Pull request by brian-brazil**
@beorn7 

I didn't add linearBuckets/exponentialBuckets as Python has range() and list comprehensions are easy.

### CODE REVIEW

File: prometheus_client/__init__.py [hunk 1]
```diff
@@ -300,6 +304,73 @@ def _samples(self):
...
+  elif d == _MINUS_INF:
+    return '-Inf'
+  else:
+    return repr(d)
+
+@_MetricWrapper
+class Histogram(object):
+  _type = 'histogram'
+  _reserved_labelnames = ['histogram']
+  def __init__(self, buckets=(.005, .01, .025, .05, .075, .1, .25, .5, .75, 1.0, 2.5, 5.0, 7.5, 10.0, _INF)):
```
Comment (beorn7) [inline prometheus_client/__init__.py]: This .75 in there really gives me the creeps... I think the default buckets should have a more regular spacing.

File: prometheus_client/__init__.py [hunk 2]
```diff
@@ -300,6 +304,73 @@ def _samples(self):
...
+    self._sum = 0.0
+    self._lock = Lock()
+    buckets = [float (b) for b in buckets]
+    if buckets != sorted(buckets):
+      # This is probably an error on the part of the user,
+      # so raise rather than sorting for them.
+      raise ValueError('Buckets not in sorted order')
+    if not buckets:
+      raise ValueError('Must have at least one bucket')
```
Comment (beorn7) [inline prometheus_client/__init__.py]: But what if that one bucket is _INF?

I'd add the check after adding the _INF and then make sure there are at least two buckets.

**Comment by beorn7** [comment]
:+1: otherwise.

File: prometheus_client/__init__.py (hunk 2 above)
Comment (brian-brazil) [inline prometheus_client/__init__.py]: Done

**Pull request merged**